File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private long cur = Long.MIN_VALUE;
    
    public boolean isValidBST(TreeNode root) {
        if(root == null) return true;
        return inOrder(root);
    }
    
    private boolean inOrder(TreeNode node){
        if(node == null) return true;
        
        if(!inOrder(node.left)) return false;
        if(node.val <= cur) return false;
        cur = node.val;
        
        return inOrder(node.right);
    }
}

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function(root) {
    if(root == null) return true;
    let cur = -Infinity;
    
    const inOrder = (node) =>{
        if (!node) return true;

        if (!inOrder(node.left)) return false;

        if (node.val <= cur) return false;
        cur = node.val;

        return inOrder(node.right);
    }
    return inOrder(root);
};